syne_tune/optimizer/schedulers/multiobjective/random_scalarization.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, Optional, Iterable, List, Callable
+
+import numpy as np
+import scipy.stats
+from numpy.random import RandomState
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes import (
+    Predictor,
+    ScoringFunction,
+)
+from syne_tune.optimizer.schedulers.searchers.utils.common import Configuration
+
+
+class MultiObjectiveLCBRandomLinearScalarization(ScoringFunction):
+    """
+    Note: This is the multi objective random scalarization scoring function based on the work of Biswajit et al. [1].
+    This scoring function uses Lower Confidence Bound as the acquisition for the scalarized objective
+    :math:`h(\mu, \sigma) = \mu - \kappa * \sigma`
+
+        | [1] Paria, Biswajit, Kirthevasan Kandasamy and Barnabás Póczos.
+        | A Flexible Framework for Multi-Objective Bayesian Optimization using Random Scalarizations.
+        | Conference on Uncertainty in Artificial Intelligence (2018).
+
+    :param predictor: Surrogate predictor for statistics of predictive distribution
+    :param weights_sampler: Callable that can generate weights for each objective.
+        Once called it will return a dictionary mapping metric name to scalarization weight as
+        {
+            <name of metric 1> : <weight for metric 1>,
+            <name of metric 2> : <weight for metric 2>,
+            ...
+        }
+    :param kappa: Hyperparameter used for the LCM portion of the scoring
+    """
```

Comment:
Add comment about `random_seed`, and state this is used only if `weights_sampler` is not given